packages/hoppscotch-data/src/graphql/index.ts

```diff
@@ -1,51 +1,73 @@
-import { HoppGQLAuth } from "./HoppGQLAuth"
+import { InferredEntity, createVersionedEntity } from "verzod"
+import V1_VERSION from "./v/1"
+import V2_VERSION from "./v/2"
 
-export * from "./HoppGQLAuth"
+export { GQLHeader } from "./v/1"
+export {
+  HoppGQLAuth,
+  HoppGQLAuthAPIKey,
+  HoppGQLAuthBasic,
+  HoppGQLAuthBearer,
+  HoppGQLAuthNone,
+  HoppGQLAuthOAuth2
+} from "./v/2"
 
 export const GQL_REQ_SCHEMA_VERSION = 2
 
-export type GQLHeader = {
-  key: string
-  value: string
-  active: boolean
-}
+export const HoppGQLRequest = createVersionedEntity({
+  latestVersion: 2,
+  versionMap: {
+    1: V1_VERSION,
+    2: V2_VERSION
+  },
+  getVersion(x) {
+    return typeof x === "object"
+      && x !== null
+      && "v" in x
+      && typeof x.v === "number"
+        ? x.v
+        : 0
+  }
```

Comment:
anyway we're using zod in `hoppscotch-data`. so why don't we use it here also ? makes it more readable and explicit. 

```suggestion
  getVersion(x) {
    const versionedObject = z.object({
      v: z.number(),
    })

    const result = versionedObject.safeParse(x)

    return result.success ? result.data.v : 0
  },
```